### PR TITLE
Server-side panels

### PR DESCRIPTION
**Pull request by forman**
This PR allows extending xcube Viewer by custom panels defined in xcube Server.

Added experimental feature that allows for extending the xcube Viewer user interface by _server-side panels_. For this to work, users can now configure xcube Server to load one or more Python modules that provide `xcube.webapi.viewer.contrib.Panel` UI-contributions. Panel instances provide two decorators `layout()` and `callback()` which are used to implement the UI and the interaction behaviour, respectively. The functionality is provided by the <https://github.com/bcdev/chartlets> Python library. A working example can be found in `examples/serve/panels-demo`.

Related xcube Viewer PR: https://github.com/xcube-dev/xcube-viewer/pull/438

Checklist:

* [x] Add unit tests and/or doctests in docstrings
* [x] Add docstrings and API docs for any new/modified user-facing classes and functions
* ~[ ] New/modified features documented in `docs/source/*`~
* [x] Changes documented in `CHANGES.md`
* [x] GitHub CI passes
* [x] AppVeyor CI passes
* [x] Test coverage remains or increases (target 100%)

### CODE REVIEW

**Review by b-yogesh — APPROVED** [review]
Overall looks good! Approved.

File: xcube/webapi/viewer/context.py [hunk 1]
```diff
@@ -2,18 +2,50 @@
...
+        if viewer_config:
+            augmentation: dict | None = viewer_config.get("Augmentation")
+            if augmentation:
+                extension_refs: list[str] = augmentation["Extensions"]
+                path: Path | None = None
+                if "Path" in augmentation:
+                    path = Path(augmentation["Path"])
+                    if not path.is_absolute():
+                        path = Path(self.base_dir) / path
+                with prepend_sys_path(path):
```
Comment (pont-us) [inline xcube/webapi/viewer/context.py:43]: I feel as though there ought to be a more elegant way to do this without changing the system path globally, but on closer examination it seems that alternative approaches would need quite a lot of import system surgery :(.

File: examples/serve/panels-demo/config.yaml [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+          Variable: band_3
+          ValueRange: [0., 5000]
+
+ServiceProvider:
+  ProviderName: "Brockmann Consult GmbH"
+  ProviderSite: "https://www.brockmann-consult.de"
+  ServiceContact:
+    IndividualName: "Norman Fomferra"
+    PositionName: "Senior Software Engineer"
+    ContactInfo:
```
Comment (AliceBalfanz) [inline examples/serve/panels-demo/config.yaml:114]: This info is outdated, pleas update with the following:
```suggestion
        ContactInfo:
              Phone:
                Voice: +49 40 696 389 300
                Facsimile: +49 40 696 389 399
              Address:
                DeliveryPoint: Chrysanderstr. 1
                City: Hamburg
                AdministrativeArea: Hamburg
                PostalCode: 21029
                Country: Germany
                ElectronicMailAddress: info@brockmann-consult.de
```

**Review by pont-us — APPROVED** [review]
Makes sense, looks good! Just a couple of tiny suggestions.

**Review by AliceBalfanz — APPROVED** [review]
Tested from a user perspective. Works well :) Minor comment on outdated info in config - but thats not crucial for merging.

File: examples/serve/panels-demo/config.yaml (hunk 2 above)
Comment (forman) [reply examples/serve/panels-demo/config.yaml:114]: Sure :)

File: xcube/webapi/viewer/context.py (hunk 1 above)
Comment (forman) [reply xcube/webapi/viewer/context.py:43]: If you know anything better, please report back to me!

**Pull request merged**